app/src/main/java/com/android/solvit/provider/ui/profile/ProviderProfile.kt

```diff
@@ -71,508 +69,258 @@ import com.android.solvit.shared.ui.navigation.Screen
-    fullName = provider.name
-    companyName = provider.companyName
-    service = Services.format(provider.service)
-    phone = provider.phone
-    location = provider.location.name
-  }
-
-  Row(modifier = Modifier.fillMaxWidth()) {
+  // Display the profile information if it's available
+  Scaffold(
+      topBar = { ProfileTopBar(navigationActions, onLogout = { showLogoutDialog = true }) },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { navigationActions.navigateTo(it.route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION_PROVIDER,
+            selectedItem = Screen.PROVIDER_PROFILE)
+      },
+      containerColor = Color.Transparent,
+  ) { paddingValues ->
     Column(
         modifier =
-            Modifier.background(colorScheme.primaryContainer)
-                .wrapContentHeight()
-                .padding(8.dp)
-                .weight(1f),
+            Modifier.fillMaxSize()
+                .padding(paddingValues)
+                .padding(horizontal = 16.dp)
+                .verticalScroll(rememberScrollState())
+                .testTag("ProfileContent"),
         horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.Top) {
-          Column(
-              modifier = Modifier.align(Alignment.Start),
-              horizontalAlignment = Alignment.Start,
-              verticalArrangement = Arrangement.Bottom) {
-                Box {
-                  IconButton(
-                      onClick = { navigationActions.goBack() },
-                      modifier = Modifier.testTag("backButton")) {
-                        Icon(
-                            Icons.AutoMirrored.Filled.ArrowBack,
-                            contentDescription = "GoBackButton",
-                            modifier = Modifier.size(24.dp),
-                            tint = colorScheme.error)
-                      }
-                }
+        verticalArrangement = Arrangement.spacedBy(16.dp)) {
+
+          // profile info card
+          ProfileInfoCard(
+              fullName = provider?.name ?: "",
+              email = provider?.companyName ?: "",
+              imageUrl = provider?.imageUrl ?: "",
+              onEdit = { navigationActions.navigateTo(Screen.PROVIDER_MODIFY_PROFILE) })
+
+          // additional infos cards
+          Card(
+              modifier = Modifier.fillMaxWidth().testTag("AdditionalInfosCard"),
+              colors = CardDefaults.cardColors(containerColor = colorScheme.background),
+              elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)) {
+                Column(
+                    verticalArrangement = Arrangement.spacedBy(8.dp),
+                    modifier = Modifier.padding(16.dp)) {
+                      ProviderItem(
+                          icon = Icons.Default.Build,
+                          text = provider?.service?.name ?: "",
+                          tag = "ServiceItem")
```

Comment:
Coud you use the for `Services.format(...)` to better display the provider's service type?